operations/jsonnet/microservices/generator.libsonnet

```diff
@@ -54,8 +72,36 @@
       volume.fromConfigMap(tempo_config_volume, $.tempo_metrics_generator_configmap.metadata.name),
       volume.fromConfigMap(tempo_overrides_config_volume, $._config.overrides_configmap_name),
       volume.fromEmptyDir(tempo_generator_wal_volume),
-    ]),
+    ])
+  ,
+
+  newGeneratorStatefulSet(name, container, with_anti_affinity=true)::
+    statefulset.new(
+      name,
+      $._config.metrics_generator.replicas,
+      $.tempo_metrics_generator_container,
+      self.tempo_metrics_generator_pvc,
+      {
+        app: target_name,
+        [$._config.gossip_member_label]: 'true',
+      },
+    )
+    + kausal.util.antiAffinityStatefulSet
+    + statefulset.mixin.spec.withServiceName(target_name)
+    + statefulset.mixin.spec.template.metadata.withAnnotations({
+      config_hash: std.md5(std.toString($.tempo_metrics_generator_configmap.data['tempo.yaml'])),
+    })
+    + statefulset.mixin.spec.template.spec.withVolumes([
+      volume.fromConfigMap(tempo_config_volume, $.tempo_metrics_generator_configmap.metadata.name),
+      volume.fromConfigMap(tempo_overrides_config_volume, $._config.overrides_configmap_name),
+    ]) +
+    statefulset.mixin.spec.withPodManagementPolicy('Parallel') +
+    statefulset.mixin.spec.template.spec.withTerminationGracePeriodSeconds(1200) +
```

Comment:
20 minutes seems kind of long. What is the rationale for this number?

Reply:
I've dropped this line to take the default, which looks like 30 seconds.  Do you think we need to wait longer than that?

Reply:
No that seems sufficient.